src/main/groovy/wooga/gradle/unity/tasks/SetAPICompatibilityLevel.groovy

```diff
@@ -38,21 +38,26 @@ class SetAPICompatibilityLevel extends ConventionTask {
             return null
         }
 
+        if (value instanceof Map){
+            return value
```

Comment:
The problem here is that you assume that the provided map is of the right type.

Reply:
Aye, will check here.